kedro/extras/datasets/svmlight/svmlight_dataset.py

```diff
@@ -0,0 +1,166 @@
+"""``SVMLightDataSet`` loads/saves data from/to a svmlight/libsvm file using an
+underlying filesystem (e.g.: local, S3, GCS). It uses sklearn functions
+``dump_svmlight_file`` to save and ``load_svmlight_file`` to load a file.
+"""
+from copy import deepcopy
+from pathlib import PurePosixPath
+from typing import Any, Dict, Optional, Tuple, Union
+
+import fsspec
+from numpy import ndarray
+from scipy.sparse.csr import csr_matrix
+from sklearn.datasets import dump_svmlight_file, load_svmlight_file
+
+from kedro.io.core import (
+    AbstractVersionedDataSet,
+    DataSetError,
+    Version,
+    get_filepath_str,
+    get_protocol_and_path,
+)
+
+# NOTE: kedro.extras.datasets will be removed in Kedro 0.19.0.
+# Any contribution to datasets should be made in kedro-datasets
+# in kedro-plugins (https://github.com/kedro-org/kedro-plugins)
+
+# Type of data input
+_DI = Tuple[Union[ndarray, csr_matrix], ndarray]
+# Type of data output
+_DO = Tuple[csr_matrix, ndarray]
+
+
+class SVMLightDataSet(AbstractVersionedDataSet[_DI, _DO]):
+    """``SVMLightDataSet`` loads/saves data from/to a svmlight/libsvm file using an
+    underlying filesystem (e.g.: local, S3, GCS). It uses sklearn functions
+    ``dump_svmlight_file`` to save and ``load_svmlight_file`` to load a file.
+
+    Data is loaded as a tuple of features and labels. Labels is NumPy array,
+    and features is Compressed Sparse Row matrix.
+
+    This format is a text-based format, with one sample per line. It does
+    not store zero valued features hence is suitable for sparse dataset.
```

Comment:
```suggestion
    not store zero valued features hence it is suitable for sparse datasets.
```

Reply:
Fixed in d8ebc47c0827adde97bb94c70f5b7906fdd6a54b